### PR TITLE
Update smart source

### PR DESCRIPTION
**Pull request by pinakidas**
Update the smart_source to consider both `DBT_RUN_ENV` and `DBT_USE_ZDEV_DATASET` while populating the tables during dbt runs

### CODE REVIEW

**Review by drnielsen — CHANGES_REQUESTED** [review]
We must keep backwards compatibility with the current behavior as default.

I think if you set a default on the env var call, it won't error out if it doesn't exist.

Otherwise, we need to change the macro signature to add a new param that defaults to the current behavior.

**Comment by pinakidas** [comment]
> We must keep backwards compatibility with the current behavior as default.
> 
> I think if you set a default on the env var call, it won't error out if it doesn't exist.
> 
> Otherwise, we need to change the macro signature to add a new param that defaults to the current behavior.

Let me try checking that

**Comment by pinakidas** [comment]
@drnielsen - updated the code changes

File: README.md [hunk 1]
```diff
@@ -45,13 +45,21 @@ To use, add a macro to your macros directory with the following contents:
 ```
 
 #### smart_source
+
 This macro helps us keep a check on the bigquery costs and at the same time validate sql queries end to end.
 You can use the codegen package to have the script generated and then replace `source` with `smart_source`.
 [Link](https://github.com/dbt-labs/dbt-codegen#usage-1) on usage of codegen for generating SQL for a base model
 
+> **_NOTE:_**  With gmi_common_dbt_utils >= 0.6.0 to populate the tables in the development environment in standard datasets
```
Comment (drnielsen) [inline README.md]: I think this would be >=0.7.0.

0.6.0 already exists. Other than that, it all looks good.

Comment (pinakidas) [reply README.md]: updated

**Review by drnielsen — APPROVED** [review]
Looks good.

**Pull request merged**